packages/smart-wallet/src/smartWallet.js

```diff
@@ -1019,6 +1072,43 @@ export const prepareSmartWallet = (baggage, shared) => {
           await E(seatRef).tryExit();
         },
       },
+
+      invoke: {
+        /**
+         * @param {string} expr
+         */
+        async evalExpr(expr) {
+          const { fromEntries } = Object;
+          trace(
+            'TODO: validate Justin syntax; see https://github.com/endojs/Jessie/pull/121#discussion_r1988126110',
```

Comment:
@michaelfig while Justin is designed to _execute_ in `O(N)`, here we'd be _parsing_ it. The PEG parser... does it backtrack? That wouldn't be O(N), would it?

_discovered in discussion with @kriskowal_

Reply:
From Bryan Ford's paper:

> The basic reason the backtracking parser can take super-linear
time is because of redundant calls to the same parse function on the
same input substring, and these redundant calls can be eliminated
through memoization.